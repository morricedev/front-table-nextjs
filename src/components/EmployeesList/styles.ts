import styled, { css } from 'styled-components';

export const EmployeesListContainer = styled.div`
  ${({ theme }) => css`
    overflow-x: auto;

    margin-top: 3.5rem;
    border: 0.1rem solid ${theme.colors.gray100};
    border-radius: 0.4rem 0.4rem 0 0;
  `}
`;

export const EmployeesTable = styled.table`
  ${({ theme }) => css`
    border-collapse: collapse;
    width: 100%;

    & tr {
      transform: scale(1);
      -webkit-transform: scale(1);
      -moz-transform: scale(1);
      box-shadow: 0 0 0.4rem rgba(0, 0, 0, 0.2);
      -webkit-box-shadow: 0 0 0.4rem rgba(0, 0, 0, 0.2);
      -moz-box-shadow: 0 0 0.4rem rgba(0, 0, 0, 0.2);
    }

    & thead th {
      padding: 1.5rem 0;

      background-color: ${theme.colors.primary};
      color: ${theme.colors.white};

      font-weight: 500;
      text-transform: uppercase;
    }

    th,
    td {
      border-bottom: 0.1rem solid ${theme.colors.gray100};
      text-align: center;
      font-size: 1.6rem;
    }

    td {
      background-color: ${theme.colors.white};
      padding: 0.5rem 1rem;
      font-weight: 400;
      font-color: ${theme.colors.black};

      & img {
        height: 3.5rem;
        width: 3.5rem;
        border-radius: 100%;
      }
    }

    @media only screen and ${theme.breakpoints.small} {
      & td {
        padding: 0.8rem 2rem;
      }

      & thead th {
        padding: 0.8rem;
      }
    }
  `}
`;
