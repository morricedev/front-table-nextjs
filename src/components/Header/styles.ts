import styled, { css } from 'styled-components';

export const Header = styled.header`
  ${({ theme }) => css`
    background-color: ${theme.colors.black};
    height: 6rem;
    display: flex;
    align-items: center;
    box-shadow: 0 0.2rem 0.8rem rgba(0, 0, 0, 0.2);
  `}
`;

export const Logo = styled.img`
  margin-left: 3.2rem;
`;
