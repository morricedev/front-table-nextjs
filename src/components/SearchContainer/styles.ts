import styled, { css } from 'styled-components';

export const SearchContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;

    @media only screen and ${theme.breakpoints.small} {
      & {
        flex-direction: column;
        align-items: center;
        gap: 2rem;
      }
    }
  `}
`;

export const Heading = styled.h1`
  ${({ theme }) => css`
    font-weight: 500;
    font-size: ${theme.font.size.medium};
  `}
`;

export const Search = styled.div`
  position: relative;
`;

export const SearchInput = styled.input`
  ${({ theme }) => css`
    width: 28.2rem;
    padding: 1.2rem 5rem 1.2rem 1.2rem;
    background-color: ${theme.colors.white};
    border: 0.1rem solid ${theme.colors.gray200};
    border-radius: 0.4rem;

    &::placeholder {
      font-family: ${theme.font.family.secondary};
      color: ${theme.colors.gray300};
    }
  `}
`;

export const SearchButton = styled.button`
  ${({ theme }) => css`
    position: absolute;
    top: 9%;
    right: 0.3rem;
    z-index: 10;
    border: none;
    background-color: transparent;
    outline: none;
    cursor: pointer;
    padding: 1rem;

    color: ${theme.colors.gray400};

    &:disabled {
      cursor: not-allowed;
    }

    svg {
      width: 17px;
      height: 17px;
    }
  `}
`;

export const ResetSearchButton = styled(SearchButton)`
  top: 8%;
  right: 2.5rem;
`;
