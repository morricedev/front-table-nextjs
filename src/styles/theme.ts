export const theme = {
  colors: {
    primary: '#5984c0',
    white: '#fff',
    bg: '#f0f0f0',
    black: '#1c1c1c',
    gray100: '#ddd',
    gray200: '#cdcdcd',
    gray300: '#9e9e9e',
    gray400: '#7d7d7d',
  },
  font: {
    family: {
      primary: '"Roboto", sans-serif',
      secondary: '"Open Sans", sans-serif',
    },
    size: {
      small: '1.6rem',
      medium: '2.4rem',
    },
  },
  breakpoints: {
    small: '(max-width: 57.6rem)',
  },
};
