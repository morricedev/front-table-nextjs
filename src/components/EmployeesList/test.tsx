import { screen } from '@testing-library/react';
import { EmployeesList } from '.';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock';

describe('<EmployeesList />', () => {
  it('should render table with not found message', () => {
    const { container } = renderTheme(<EmployeesList employees={[]} />);

    expect(
      screen.getByText('Funcionários não foram encontrados'),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render table with three employees', () => {
    const { container } = renderTheme(<EmployeesList employees={mock} />);

    const tbodyTrs = container.querySelector('tbody').querySelectorAll('tr');

    expect(
      screen.queryByText('Funcionários não foram encontrados'),
    ).not.toBeInTheDocument();
    expect(tbodyTrs).toHaveLength(mock.length);
    expect(container).toMatchSnapshot();
  });
});
