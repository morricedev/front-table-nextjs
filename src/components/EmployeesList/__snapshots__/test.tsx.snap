// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployeesList /> should render table with not found message 1`] = `
.c0 {
  overflow-x: auto;
  margin-top: 3.5rem;
  border: 0.1rem solid #ddd;
  border-radius: 0.4rem 0.4rem 0 0;
}

.c1 {
  border-collapse: collapse;
  width: 100%;
}

.c1 tr {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  box-shadow: 0 0 0.4rem rgba(0,0,0,0.2);
  -webkit-box-shadow: 0 0 0.4rem rgba(0,0,0,0.2);
  -moz-box-shadow: 0 0 0.4rem rgba(0,0,0,0.2);
}

.c1 thead th {
  padding: 1.5rem 0;
  background-color: #5984c0;
  color: #fff;
  font-weight: 500;
  text-transform: uppercase;
}

.c1 th,
.c1 td {
  border-bottom: 0.1rem solid #ddd;
  text-align: center;
  font-size: 1.6rem;
}

.c1 td {
  background-color: #fff;
  padding: 0.5rem 1rem;
  font-weight: 400;
  font-color: #1c1c1c;
}

.c1 td[colspan='5'] {
  padding: 1.5rem;
}

.c1 td img {
  height: 3.5rem;
  width: 3.5rem;
  border-radius: 100%;
}

@media only screen and (max-width:57.6rem) {
  .c1 td {
    padding: 0.8rem 2rem;
  }

  .c1 thead th {
    padding: 0.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <thead>
        <tr>
          <th>
            Foto
          </th>
          <th>
            Nome
          </th>
          <th>
            Cargo
          </th>
          <th>
            Data de admissão
          </th>
          <th>
            Telefone
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            colspan="5"
          >
            Funcionários não foram encontrados
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<EmployeesList /> should render table with three employees 1`] = `
.c0 {
  overflow-x: auto;
  margin-top: 3.5rem;
  border: 0.1rem solid #ddd;
  border-radius: 0.4rem 0.4rem 0 0;
}

.c1 {
  border-collapse: collapse;
  width: 100%;
}

.c1 tr {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  box-shadow: 0 0 0.4rem rgba(0,0,0,0.2);
  -webkit-box-shadow: 0 0 0.4rem rgba(0,0,0,0.2);
  -moz-box-shadow: 0 0 0.4rem rgba(0,0,0,0.2);
}

.c1 thead th {
  padding: 1.5rem 0;
  background-color: #5984c0;
  color: #fff;
  font-weight: 500;
  text-transform: uppercase;
}

.c1 th,
.c1 td {
  border-bottom: 0.1rem solid #ddd;
  text-align: center;
  font-size: 1.6rem;
}

.c1 td {
  background-color: #fff;
  padding: 0.5rem 1rem;
  font-weight: 400;
  font-color: #1c1c1c;
}

.c1 td[colspan='5'] {
  padding: 1.5rem;
}

.c1 td img {
  height: 3.5rem;
  width: 3.5rem;
  border-radius: 100%;
}

@media only screen and (max-width:57.6rem) {
  .c1 td {
    padding: 0.8rem 2rem;
  }

  .c1 thead th {
    padding: 0.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <table
      class="c1"
    >
      <thead>
        <tr>
          <th>
            Foto
          </th>
          <th>
            Nome
          </th>
          <th>
            Cargo
          </th>
          <th>
            Data de admissão
          </th>
          <th>
            Telefone
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <img
              alt="João"
              src="https://img.favpng.com/25/7/23/computer-icons-user-profile-avatar-image-png-favpng-LFqDyLRhe3PBXM0sx2LufsGFU.jpg"
            />
          </td>
          <td>
            João
          </td>
          <td>
            Back-end
          </td>
          <td>
            01/12/2019
          </td>
          <td>
            +55 (51) 23456-7890
          </td>
        </tr>
        <tr>
          <td>
            <img
              alt="Roberto"
              src="https://e7.pngegg.com/pngimages/550/997/png-clipart-user-icon-foreigners-avatar-child-face.png"
            />
          </td>
          <td>
            Roberto
          </td>
          <td>
            Front-end
          </td>
          <td>
            11/03/2020
          </td>
          <td>
            +55 (50) 32165-4789
          </td>
        </tr>
        <tr>
          <td>
            <img
              alt="Maria"
              src="https://www.clipartmax.com/png/middle/277-2772117_user-profile-avatar-woman-icon-avatar-png-profile-icon.png"
            />
          </td>
          <td>
            Maria
          </td>
          <td>
            Front-end
          </td>
          <td>
            14/03/2020
          </td>
          <td>
            +55 (57) 89456-1230
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
